From 9d1ba9fdc93b386e5e2546e6a1ee66e086ea1fef Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Sun, 14 Apr 2019 23:52:52 +0200
Subject: [PATCH 3/3] Updater: Make upgrade-url device specific

* After moving from the generic /upgrade.html url to a device specific
  one, updater needs to display that instead

Change-Id: If9fc6bddfab206a97283f2080ea4df0cbb742a7d
---
 res/values/strings.xml                            | 2 +-
 src/org/lineageos/updater/UpdatesListAdapter.java | 2 +-
 src/org/lineageos/updater/misc/Utils.java         | 6 ++++++
 3 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index 6ce20bb..34f2fd3 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -144,7 +144,7 @@
 
     <string name="blocked_update_dialog_title">Update blocked</string>
     <string name="blocked_update_dialog_message">This update cannot be installed using the updater app.  Please read <xliff:g id="info_url">%1$s</xliff:g> for more information.</string>
-    <string name="blocked_update_info_url" translatable="false">http://wiki.lineageos.org/upgrading.html</string>
+    <string name="blocked_update_info_url" translatable="false">http://wiki.lineageos.org/devices/<xliff:g id="device_name">%1$s</xliff:g>/upgrade</string>
 
     <string name="export_channel_title">Export completion</string>
     <string name="new_updates_channel_title">New updates</string>
diff --git a/src/org/lineageos/updater/UpdatesListAdapter.java b/src/org/lineageos/updater/UpdatesListAdapter.java
index e6f4e9a..a1aa944 100644
--- a/src/org/lineageos/updater/UpdatesListAdapter.java
+++ b/src/org/lineageos/updater/UpdatesListAdapter.java
@@ -536,7 +536,7 @@ public class UpdatesListAdapter extends RecyclerView.Adapter<UpdatesListAdapter.
     private void showInfoDialog() {
         String messageString = String.format(StringGenerator.getCurrentLocale(mActivity),
                 mActivity.getString(R.string.blocked_update_dialog_message),
-                mActivity.getString(R.string.blocked_update_info_url));
+                Utils.getUpgradeBlockedURL(mActivity));
         SpannableString message = new SpannableString(messageString);
         Linkify.addLinks(message, Linkify.WEB_URLS);
         AlertDialog dialog = new AlertDialog.Builder(mActivity)
diff --git a/src/org/lineageos/updater/misc/Utils.java b/src/org/lineageos/updater/misc/Utils.java
index 9299114..eed4d56 100644
--- a/src/org/lineageos/updater/misc/Utils.java
+++ b/src/org/lineageos/updater/misc/Utils.java
@@ -166,6 +166,12 @@ public class Utils {
                 .replace("{incr}", incrementalVersion);
     }
 
+    public static String getUpgradeBlockedURL(Context context) {
+        String device = SystemProperties.get(Constants.PROP_NEXT_DEVICE,
+                SystemProperties.get(Constants.PROP_DEVICE));
+        return context.getString(R.string.blocked_update_info_url, device);
+    }
+
     public static String getChangelogURL(Context context) {
         String device = SystemProperties.get(Constants.PROP_NEXT_DEVICE,
                 SystemProperties.get(Constants.PROP_DEVICE));
-- 
2.17.1

